### PR TITLE
Refactoring cfile.c and arena.c

### LINKED ISSUE #2588: Failed to specify which Dex to parse in a multidex APK
Hi there!

I currently use Rizin to analyze a multidex APK. I chose [this sample](https://github.com/rizinorg/rizin/files/8618944/c02bd4289acacf65d9cb70ce42d0ff27.zip) (Password: infected) as my target and tried to get data from all its Dex files respectively with command `ou`. However, an error occurred.

### Work environment

| Questions                                             | Answers                                                      |
| ----------------------------------------------------- | ------------------------------------------------------------ |
| OS/arch/bits (mandatory)                              | Ubuntu 22.04                                                 |
| File format of the file you reverse (mandatory)       | APK                                                          |
| Architecture/bits of the file (mandatory)             | Dalvik                                                       |
| `rizin -v` full output, **not truncated** (mandatory) | rizin 0.4.0-git @ linux-x86-64<br/>commit: d8ed585cba557985fb46e10693f1f6726dd75bf2, build: 2022-05-04__02:27:18 |

### Expected behavior

Successfully switch to the specified Dex by using command `ou`.

### Actual behavior

An error occurred as follows.

```
$> rizin apk://c02bd4289acacf65d9cb70ce42d0ff27.apk 
WARNING: No calling convention defined for this file, analysis may be inaccurate.
 -- Thanks for using rizin!
[0x100966804]> ol
 3 - r-x 0x000dfd24 classes2.dex
 4 * r-x 0x0091fd3c classes.dex
 5 - r-x 0x00006eda vfile://0/reloc-targets
 6 - r-x 0x0000098e vfile://1/reloc-targets
[0x100966804]> ou 3
ERROR: Could not find any opened file with fd 3
[0x100966804]> 
```

### Steps to reproduce the behavior

- Open [this sample](https://github.com/rizinorg/rizin/files/8618944/c02bd4289acacf65d9cb70ce42d0ff27.zip) with Rizin. (Password: infected)
- Run the following commands.

```
ol
ou 3
```

### PR DESCRIPTION
**Pull request by wargio**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

Refactoring of cfile.c and arena.c

Fix #2588

### CODE REVIEW

File: librz/core/cfile.c [hunk 1]
```diff
@@ -902,189 +906,247 @@ static bool map_multi_dex(RzCore *core, RzIODesc *desc, ut32 id) {
 	return true;
 }
 
-RZ_API bool rz_core_bin_load(RZ_NONNULL RzCore *r, RZ_NULLABLE const char *filenameuri, ut64 baddr) {
-	rz_return_val_if_fail(r, false);
+static bool is_apk_uri_with_file_path(const char *uri) {
+	return rz_str_startswith(uri, "apk://") && (strstr(uri + 6, ".apk//") || strstr(uri + 6, ".zip//"));
```
Comment (ret2libc) [inline librz/core/cfile.c:910]: Why the checks for `"./apk/"` and `".zip/"`? I think the initial `apk://` should be always present and only that should determine the plugin.

Comment (ret2libc) [reply librz/core/cfile.c:910]: Apart from this.... why do we even want this function here? This is RzCore, there should be no reference to specific plugin nor any plugin-specific logic.

File: librz/reg/arena.c [hunk 2]
```diff
@@ -311,35 +311,33 @@ RZ_API ut8 *rz_reg_arena_dup(RzReg *reg, const ut8 *source) {
...
-		return -1;
-	}
-	int bin_str_len = (len + 1) / 2; // 2 hex chrs for 1 byte
-	ut8 *bin_str = malloc(bin_str_len);
-	if (!bin_str) {
-		eprintf("Failed to decode hex str.\n");
-		return -1;
-	}
-	rz_hex_str2bin(str, bin_str);
+RZ_API bool rz_reg_arena_set_bytes(RzReg *reg, const char *hex) {
```
Comment (ret2libc) [inline librz/reg/arena.c:314]: doxygen doc

File: librz/core/cfile.c (hunk 1 above)
Comment (wargio) [reply librz/core/cfile.c:910]: because apks are zips, and loading an apk with apk://file.zip should be allowed

Comment (wargio) [reply librz/core/cfile.c:910]: this is to fix the loading of multidex which has to happen here, in the core.

**Pull request closed without merging**